datadog/fwprovider/resource_datadog_app.go

```diff
@@ -0,0 +1,292 @@
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	frameworkPath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &appResource{}
+	_ resource.ResourceWithImportState = &appResource{}
+)
+
+type appResource struct {
+	Api  *datadogV2.AppBuilderApi
+	Auth context.Context
+}
+
+// try single property JSON input -> validation will be handled on the API side
+type appResourceModel struct {
+	ID      types.String         `tfsdk:"id"`
+	AppJson jsontypes.Normalized `tfsdk:"app_json"`
+}
+
+func NewAppResource() resource.Resource {
+	return &appResource{}
+}
+
+func (r *appResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetAppBuilderApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *appResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "app"
+}
+
+func (r *appResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Provides a Datadog App resource. This can be used to create and manage a Datadog App from the App Builder product.",
+		Attributes: map[string]schema.Attribute{
+			"id": utils.ResourceIDAttribute(),
+			"app_json": schema.StringAttribute{
+				Required:    true,
+				Description: "The JSON representation of the App.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+				CustomType: jsontypes.NormalizedType{},
+			},
+		},
+	}
+}
+
+func (r *appResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *appResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var plan appResourceModel
+	diags := request.Plan.Get(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	createRequest, err := appModelToCreateApiRequest(plan)
+	if err != nil {
+		response.Diagnostics.AddError("Error building create app request", err.Error())
+		return
+	}
+
+	resp, httpResp, err := r.Api.CreateApp(r.Auth, *createRequest)
+	if err != nil {
+		if httpResp != nil {
+			// error body may have useful info for the user
```

Comment:
just an observation from my testing - you should make sure your API doesn't expose any internal errors or they will show up in TF errors as well